src/slapper/Main.php

```diff
@@ -670,10 +670,18 @@ private function makeNBT($type, Player $player) {
 		$nbt->Commands = new CompoundTag("Commands", []);
 		$nbt->MenuName = new StringTag("MenuName", "");
 		$nbt->SlapperVersion = new StringTag("SlapperVersion", $this->getDescription()->getVersion());
-		if($type === "Human") {
+
+		// TODO: This will need to be updated when PMMP updates Human class to handle Capes and Custom Geometry
+		if($type === "Human"){
 			$player->saveNBT();
 			$nbt->Inventory = clone $player->namedtag->Inventory;
-			$nbt->Skin = new CompoundTag("Skin", ["Data" => new StringTag("Data", $player->getSkin()->getSkinData()), "Name" => new StringTag("Name", $player->getSkin()->getSkinId())]);
+			$nbt->Skin = new CompoundTag("Skin", [
```

Comment:
This dynamic field assignment should be avoided. This is deprecated and will.be removed in the future.

Reply:
You've highlighted 678 where Skin is assigned but this same method is happening all the way through makeNBT with Pos, Motion, Rotation, Health, Commands, MenuName, SlapperVersion, and Inventory.  Would it not make sense to adopt the existing method to resolve the Geometry problem and then create another PR to solve the 'dynamic assignment' issue as this will require a rewrite of makeNBT and possible other areas?

Reply:
This comment was made here for reference purposes.

Reply:
@dktapps I need you to advise on my reply before I tackle changes for this.